### PR TITLE
Upgrade to Ogre 13

### PR DESCRIPTION
**Pull request by only-a-ptr**
Closes https://github.com/RigsOfRods/rigs-of-rods/issues/2599
Closes https://github.com/RigsOfRods/rigs-of-rods/issues/2351
Fixes https://github.com/RigsOfRods/rigs-of-rods/issues/2823
Fixes https://github.com/RigsOfRods/rigs-of-rods/issues/2353

### CODE REVIEW

**Comment by tritonas00** [comment]
Here segfaults at

```
Thread 1 "RoR" received signal SIGSEGV, Segmentation fault.
0x00007ffff79e62f2 in Ogre::SceneNode::setFixedYawAxis (this=0x0, useFixed=useFixed@entry=false, fixedAxis=...) at /home/babis/Downloads/ogre13/ror-dependencies/Source/ogre/OgreMain/src/OgreSceneNode.cpp:402
402	        mYawFixed = useFixed;
```

```
#0  0x00007ffff79e62f2 in Ogre::SceneNode::setFixedYawAxis (this=0x0, useFixed=useFixed@entry=false, fixedAxis=...)
    at /home/babis/Downloads/ogre13/ror-dependencies/Source/ogre/OgreMain/src/OgreSceneNode.cpp:402
#1  0x000055555564aa6b in RoR::GfxEnvmap::SetupEnvMap (this=0x555555bdd520 <RoR::App::g_gfx_scene+128>)
    at /home/babis/Downloads/ogre13/ror-dependencies/rigs-of-rods/source/main/gfx/EnvironmentMap.cpp:51
#2  0x00005555555e1fc5 in main (argc=<optimized out>, argv=<optimized out>)
    at /home/babis/Downloads/ogre13/ror-dependencies/rigs-of-rods/source/main/main.cpp:184
    ```

**Comment by paroj** [comment]
> SceneNode::setFixedYawAxis (**this=0x0**, ...

that camera is not attached to a scene node

**Comment by tritonas00** [comment]
> > SceneNode::setFixedYawAxis (**this=0x0**, ...
> 
> that camera is not attached to a scene node

Right, fixed. Next stop the `Parameter called worldViewProj does not exist` thing, and i can't really figure it out. Probably Petr can fix it.

**Comment by paroj** [comment]
you can look in the debugger for the material name, which should help.

Also, there is https://ogrecave.github.io/ogre/api/latest/class_ogre_1_1_gpu_program_parameters.html#a436cd5b3546b6e492f1bc293bd2314d9
if you want to get it running quickly

**Comment by tritonas00** [comment]
> you can look in the debugger for the material name, which should help.
> 
> Also, there is https://ogrecave.github.io/ogre/api/latest/class_ogre_1_1_gpu_program_parameters.html#a436cd5b3546b6e492f1bc293bd2314d9 if you want to get it running quickly

Tried Debug build, finally found it, thank you!

kk

**Comment by tritonas00** [comment]
Next stop:

```
ItemIdentityException: Could not find font CyberbitEnglish in TextAreaOverlayElement::setFontName at /home/babis/Downloads/ogre13/ror-dependencies/Source/ogre/Components/Overlay/src/OgreTextAreaOverlayElement.cpp (line 435)
```

When loading a terrain, font exists it seems https://github.com/RigsOfRods/rigs-of-rods/search?q=CyberbitEnglish

**Comment by tritonas00** [comment]
@paroj while we are at videocameras, any ideia why this happens (for years)?

right mirror - you can see the sky
kk
left mirror - black sky
kkk

**Comment by tritonas00** [comment]
> `getTextureUnitState("Diffuse_Map")` must be returning NULL. Not sure why though.

this https://github.com/RigsOfRods/rigs-of-rods/pull/2858/commits/75928f0aed38db71c4ccc805486a089e055fa1be fixes it, not sure if its the right approach though

File: source/main/plugins_d.cfg.in [hunk 1]
```diff
@@ -1,15 +1,16 @@
 # Defines plugins to load
+# NOTE: OGRE13 adds the _d suffixes itself!
```
Comment (paroj) [inline source/main/plugins_d.cfg.in:2]: https://github.com/OGRECave/ogre/blob/master/Docs/1.12-Notes.md#automatic-plugin-discovery-for-windows-debug-builds

**Comment by tritonas00** [comment]
There is an issue with Caelum "Time of day" slider, it goes by steps instead of smooth.
```
std::cout << time << std::endl;
std::cout << m_daytime << std::endl;
```

returns different numbers on master:
```
0.000447373
0.000466088
0.000447373
0.000466331
...
```

and with ogre 13 + latest ogrecave caelum always:
```
2.45154e+06
2.45154e+06
2.45154e+06
2.45154e+06
```

**Comment by tritonas00** [comment]
Known issues:

- full cache generation still slower than before (zzip vs zip)
-  when reflection/refraction water is selected and you load a terrain with water, you can't type or click in imgui panels, like the selector
- caelum has several issues: moon is misplaced near to the sun, "Time of day" slider goes by steps (hacked around it https://github.com/RigsOfRods/rigs-of-rods/pull/2858/commits/926dda9dab72764b527e637f2db43d736d60ad66) and extra weather (like snow) doesn't work

**Comment by paroj** [comment]
> CaelumPlugin API dropped ...

this only affects initialization:
https://github.com/OGRECave/ogre-caelum/commit/70bebc7e28bdb92d58664378056fc9705bdbbf1c#diff-8904f998aade7df6ea9a53f3dcbd5edf259fe6b8aaa7fb5179b8708fefde9773R101-R104

so that call should still work.

**Comment by tritonas00** [comment]
> > CaelumPlugin API dropped ...
> 
> this only affects initialization: [OGRECave/ogre-caelum@70bebc7#diff-8904f998aade7df6ea9a53f3dcbd5edf259fe6b8aaa7fb5179b8708fefde9773R101-R104](https://github.com/OGRECave/ogre-caelum/commit/70bebc7e28bdb92d58664378056fc9705bdbbf1c#diff-8904f998aade7df6ea9a53f3dcbd5edf259fe6b8aaa7fb5179b8708fefde9773R101-R104)
> 
> so that call should still work.

I'm getting
```
19:50:33: Parsing script RoRSkies.os
19:50:33: Error: ScriptCompiler - unexpected token in (1): 'caelum_sky_system'. If this is a legacy script you must prepend the type (e.g. font, overlay).
19:50:33: Error: ScriptCompiler - unexpected token in (7): 'point_starfield'. If this is a legacy script you must prepend the type (e.g. font, overlay).
19:50:33: Error: ScriptCompiler - unexpected token in (21): 'sun'. If this is a legacy script you must prepend the type (e.g. font, overlay).
19:50:33: Error: ScriptCompiler - unexpected token in (30): 'moon'. If this is a legacy script you must prepend the type (e.g. font, overlay).
19:50:33: Error: ScriptCompiler - unexpected token in (51): 'sky_dome'. If this is a legacy script you must prepend the type (e.g. font, overlay).
19:50:33: Error: ScriptCompiler - unexpected token in (57): 'cloud_system'. If this is a legacy script you must prepend the type (e.g. font, overlay).
19:50:33: Error: ScriptCompiler - unexpected token in (59): 'cloud_layer'. If this is a legacy script you must prepend the type (e.g. font, overlay).
```

current ror master reads it cleanly

**Comment by paroj** [comment]
where is your `CaelumPlugin::getSingleton().initialise();`? That must happen before you initialise the resource group containing the caelum script. The caelum demo successfully parses TestSkyScript.os

**Comment by tritonas00** [comment]
> where is your `CaelumPlugin::getSingleton().initialise();`? That must happen before you initialise the resource group containing the caelum script. The caelum demo successfully parses TestSkyScript.os

OK i should have done that, now it parses them correctly, still this https://github.com/RigsOfRods/rigs-of-rods/blob/49dc172b2a2ef58f519fdf2a5469af9530f22d41/source/main/gfx/SkyManager.cpp#L89 complains

```
05:34:37: [RoR] Exception while loading sky script: RuntimeAssertionException: !resourceName.empty() failed. resourceName is empty string in openResourceImpl at /home/babis/Downloads/ogre13/ror-dependencies/Source/ogre/OgreMain/src/OgreResourceGroupManager.cpp (line 528)
```

**Comment by paroj** [comment]
this means some resource got an empty name. Need a callstack to figure out which one.

**Comment by tritonas00** [comment]
@paroj any idea why `texture` doesn't work here? https://github.com/RigsOfRods/rigs-of-rods/blob/master/resources/managed_materials/managed_mats.material#L13
with `texture_alias` roads are normal, with `texture` they are black

**Comment by paroj** [comment]
> any idea why texture doesn't work here?

see my previous comment https://github.com/RigsOfRods/rigs-of-rods/pull/2858#pullrequestreview-899223118

**Comment by tritonas00** [comment]
https://github.com/RigsOfRods/rigs-of-rods/pull/2943

**Pull request closed without merging**